### PR TITLE
API tests for Customers api

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: auth.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
...
...VzdC5ubyIsImN1c3RvbTphY2Nlc3NSaWdodHMiOiJSRUFEX0RPSV9SRVFVRVNUIiwiaWF0IjoxNjE0OTI5NTc4LCJlbWFpbCI6ImFwaS10ZXN0LXVzZXJAdGVzdC5ubyIsImN1c3RvbTppZGVudGlmaWVycyI6ImZlaWRlOmFwaS10ZXN0LXVzZXJAdGVzdC5ubyIsImN1c3RvbTpjcmlzdGluSWQiOiJodHRwczpcL1wvYXBpLmNyaXN0aW4ubm9cL3YyXC9pbnN0aXR1dGlvbnNcLzExMTExMTExMTEiLCJjb2duaXRvOnVzZXJuYW1lIjoiNzA2Yzc2ZGItMzVlNy00NzRkLTk4ODgtYmRhNmZjM2RiZDUzIiwiZ2l2ZW5fbmFtZSI6IkFQSSB0ZXN0IiwiYXVkIjoiMmpwaWxjNjRoYjI1cjFoajQxcDBoc2ZjMGwiLCJjdXN0b206Y3VzdG9tZXJJZCI6Imh0dHBzOlwvXC9hcGkuZGV2Lm52YS5hd3MudW5pdC5ub1wvY3VzdG9tZXJcLzljMWM5NDFmLWM3MTMtNDkwMC04NDgyLWY3ZDI2ODRmYzYzMCIsImV2ZW50X2lkIjoiNDI5NWU1NjYtN2VhMC00MTc3LWJjNTMtZGE3MmM3YWZjOTEzIiwidG9rZW5fdXNlIjoiaWQiLCJuYW1lIjoiQVBJIHRlc3QgVXNlciIsImN1c3RvbTpjb21tb25OYW1lIjoiQVBJIHRlc3QgVXNlciIsImZhbWlseV9uYW1lIjoiVXNlciIsImN1c3RvbTphZmZpbGlhdGlvbiI6IlttZW1iZXIsIGVtcGxveWVlLCBzdGFmZl0ifQ.nPUcGsJGG5QkF5R0fhuOKJcgKPnILfEPz0QG0V9YousNcOvaz5R_-PUdi2G7sL9dMgDuujbSmaauGra8UjPlv6D48FD8mnOHQYNpb_f3QI6COjtT8iIcfFrelmRmrnnH9wKYAq5yjn0asgpf5pwUk8ywobaySAsAyEKLD2fwltJPTSjH1XhJMRvhUa5dQg8Iz0xOxrq4X4Ixo2LiPjo3U7Afpp4SIJXNvILs33UO-BU5cThcolTpYA6IJyMa0VDeLMiMRShLXlgaY5xFBt1Dpy2AW-CGGcz-Nd6FbJeax3I6i4cwIKU70L_x0hqoqeiPzmaPcPJ4QHLrbTCLWIZEPA"
```
Comment (janbj) [inline auth.json]: How does this work? Is Cognito set up to accept a static token?

File: karate-config.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+function setup() {
+  karate.log('Testing...');
+  var auth = karate.read('../../auth.json');
+  var config = {
+    BEARER_TOKEN: auth.BEARER_TOKEN,
```
Comment (janbj) [inline karate-config.js:5]: Maybe read from env var?

File: auth.json (hunk 1 above)
Comment (Eiriknil) [reply auth.json]: I have a python script that logs in and update the Bearer token. This is not optimal, but haven't found a better solution yet

File: karate-config.js (hunk 2 above)
Comment (Eiriknil) [reply karate-config.js:5]: Tried that, was hard to find a simple solution. I will try to find a better solution.

File: login_aws.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        })
+    return response['AuthenticationResult']['IdToken']
+
+
+def find_user(email):
+    response = client.list_users(UserPoolId=USER_POOL_ID)
+    for cognito_user in response['Users']:
+        for attribute in cognito_user['Attributes']:
+            if attribute['Name'] == 'email' and attribute['Value'] == email:
+                return email
```
Comment (axthosarouris) [inline login_aws.py]: refactor to method to be clearer what you are looking for. e.g. __search_user_in_user_list__

File: login_aws.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    print('User with email {} not found'.format(email))
+    return ''
+
+
+def run():
+    username = find_user(email=TEST_USER_EMAIL)
+    if username != '':
+        bearer_token = login(username=username)
+        with open('auth.json', 'w') as outfile:
+            json.dump({'BEARER_TOKEN': bearer_token}, outfile, indent=4)
```
Comment (axthosarouris) [inline login_aws.py]: refactor to method

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: tests/customers/customers.feature [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    And match header.details == 'Bad Request'
+
+    Scenario: GET by Customer id returns status OK and Customer details
+    * path '/'
+    * method GET
+    * def customerId = findCustomer('TEST_CUSTOMER_GET_BY_ID', response.customers)
+    Given path '/' + customerId
+    When method GET
+    Then status 200
+     And match response contains getCustomerIdSuccessResponse
```
Comment (brinxmat) [inline tests/customers/customers.feature]: Indent

File: tests/registrations/registrations.feature [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Feature: Registration API tests
+
+    Scenario: GET returns Registration and status Ok when requesting existing Registration
```
Comment (brinxmat) [inline tests/registrations/registrations.feature]: Maybe not part of this PR?

File: tests/users_internal/users_internal.feature [hunk 7]
```diff
@@ -72,7 +72,7 @@ Feature: Users internal API
         And match response.title == 'Bad Request'
         And match response.detail == 'JSON object is missing a type attribute'
     
-    Scenario: PUT returns status Success when updating an existing User with correct payload
+    Scenario: PUT returns status Accepted when updating an existing User with correct payload
```
Comment (brinxmat) [inline tests/users_internal/users_internal.feature:75]: Accepted != 200

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: tests/registrations/registrations.feature (hunk 6 above)
Comment (Eiriknil) [reply tests/registrations/registrations.feature]: Yeah, forgot to switch branch there, will remove it

**Review by axthosarouris — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Review by janbj — APPROVED** [review]

**Pull request merged**